tests/dashbio_demos/app_clustergram.py

```diff
@@ -652,12 +652,14 @@ def add_marker(
                 marker['group'] = int(cluster_number)
                 marker['annotation'] = annotation
                 marker['color'] = color['hex']
+
+                current_group_markers[
+                    '{}_group_marker'.format(cluster_dimension)
+                ].append(marker)
             except ValueError:
                 pass
-
-            current_group_markers[
-                '{}_group_marker'.format(cluster_dimension)
-            ].append(marker)
+            except UnboundLocalError:
```

Comment:
Can you clarify exactly when this error would be raised? I don't think we're using any global variables within this function.

Reply:
yes basically in the if statement on line 644 `cluster_number` and `cluster_dimension` are defined if true and undefined if not true. The `UnboundLocalError` exception is required to ensure errors don't get thrown in either case.